.github/workflows/pylint.yml

```diff
@@ -0,0 +1,69 @@
+name: Linting
+on:
+  pull_request:
+    branches:
+      - '*'
+permissions:
+  pull-requests: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'
+
+      - name: Store the PR branch
+        run: |
+          echo "SHA=$(git rev-parse "$GITHUB_SHA")" >> $GITHUB_OUTPUT
+        id: git
+
+      - name: Switch to main branch
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Install tobac and pylint 
+        run: |
+          pip install .
+          pip install --upgrade pylint
```

Comment:
Would it be better to use conda/mamba to create the environment, consistent with other actions, rather than pip?

Reply:
I had some trouble getting pylint to work with the conda environment but I'll take another look.

Reply:
Ah fair, if it causes more work then it's probably not worth changing it. Seems to work fine for me via conda locally as long as it isn't using python 3.12

Also good to check that we can install tobac with pip!

Reply:
Ok, it's working now. I just forgot to specify the shell in which pylint is supposed to run.